ocaml/xapi/vm_platform.ml

```diff
@@ -166,3 +174,8 @@ let check_restricted_flags ~__context platform =
 
   if is_true nested_virt platform false
   then Pool_features.assert_enabled ~__context ~f:Features.Nested_virt
+
+let check_restricted_device_model ~__context platform =
+  if not (is_valid_device_model device_model platform) then
+    raise (Api_errors.Server_error(Api_errors.invalid_value,
+            [Printf.sprintf "platform:%s when vm has VUSBs" device_model ; try List.assoc device_model platform with _ -> ""]))
```

Comment:
Instead of returning `""` it would be better to return `(undefined)`.

The code can be slightly simplified to avoid long lines:

```
let check_restricted_device_model ~__context platform =
  if not (is_valid_device_model device_model platform) then
    raise Api_errors.(Server_error.invalid_value,
                      [ Printf.sprintf "platform:%s when vm has VUSBs" device_model 
                      ; try List.assoc device_model platform with Not_found -> "(undefined)"
                      ])
```

Reply:
good suggestion of "undefined". Thanks.

Reply:
updated